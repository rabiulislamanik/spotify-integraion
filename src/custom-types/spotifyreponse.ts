export interface ISpotifyCommonResponse {
    next:string|null,
    total:number
}

export interface IAlbums extends ISpotifyCommonResponse{
    items:IAlbumItem[]
}
  
export interface INewReleaseAlbumResponse{
    albums:IAlbums
}

export interface IAlbumItem {
    album_type:string,
    artists:{external_urls:{spotify:string},name:string}[],
    external_urls:{ spotify: string},
    name:string,
    release_date:string,
    total_tracks:number,
    images:{url:string}[]
}

export interface IPlaylists extends ISpotifyCommonResponse{
    items:IPlaylistItem[]
}

export interface IFeaturedPlaylistsResponse{
    message:string,
    playlists:IPlaylists
}


export interface IPlaylistItem {
    description:string,
    external_urls:{ spotify: string},
    name:string,
    owner:{display_name:string,external_urls:{ spotify: string}},
    images:{url:string}[]
}


export interface ICategories extends ISpotifyCommonResponse{
    items:ICategoryItem[]
}

export interface ICategoriesResponse{
    message:string,
    categories:ICategories
}

export interface ICategoryItem {
    icons:{url:string}[],
    name:string,
    id:string
}
