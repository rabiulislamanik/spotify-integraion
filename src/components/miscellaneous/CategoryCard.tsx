import React from "react";
import { ICategoryItem } from '../../custom-types/spotifyreponse';

interface ICategoryCard{
    category:ICategoryItem
}

function CategoryCard (props:ICategoryCard){
  return (
    <div className="card">
      <img className="card-image" src={props.category.icons[0].url} alt="icon" height="256" width="256"/>
      <p>{props.category.name}</p>
    </div>
  );
}

export default CategoryCard;