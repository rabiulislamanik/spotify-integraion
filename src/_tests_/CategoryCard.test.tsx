import React from 'react';
import renderer from 'react-test-renderer';

import CategoryCard from '../components/miscellaneous/CategoryCard';
import { ICategoryItem } from '../custom-types/spotifyreponse';

it('renders correctly when an Category is passed as prop', () => {
  const fakeCategory:ICategoryItem = {
    icons:[{url:'icon_url'}],
    name:'Dummy Category',
    id:'dummy_category'
  }; 
  const tree = renderer.create(<CategoryCard category={fakeCategory}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
