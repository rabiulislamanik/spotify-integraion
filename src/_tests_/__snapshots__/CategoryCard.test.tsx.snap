// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when an Category is passed as prop 1`] = `
<div
  className="card"
>
  <img
    alt="icon"
    className="card-image"
    height="256"
    src="icon_url"
    width="256"
  />
  <p>
    Dummy Category
  </p>
</div>
`;
